packages/flutter/lib/src/painting/debug_overflow_indicator.dart

```diff
@@ -0,0 +1,392 @@
+      TileMode.repeated,
+    );
+  static final Paint _debugLabelBackgroundPaint = new Paint()..color = const Color(0xFFFFFFFF);
+
+  final TextPainter _debugMarkerLabel = new TextPainter()
+    ..textDirection = TextDirection.ltr; // This label is in English.
+
+  TextDirection get textDirection => null;
+
+  RenderObject overflowRenderObject;
+
+  String get _containerName => runtimeType.toString();
+
+  /// Set to true to trigger a debug message in the console upon
+  /// the next paint call.
+  ///
+  /// Will be set to false after a call to
+  /// paint, so it must be reset each time a debug message is desired.
+  ///
+  /// Typically, this is set to true inside of the
+  /// [RenderObject.reassemble] override in the [RenderObject] that
+  /// owns this [DebugOverflowIndicatorMixin].
+  bool overflowReportNeeded = true;
+
+  /// An explanation as to what went wrong, and how to fix it.
+  ///
+  /// Override this to customize the hints given when an overflow occurs.  The
+  /// amount of the overflow in each direction, and the details of the
+  /// [RenderObject] that is overflowing will be printed along with this
+  /// message.
+  String get overflowHints {
+    return 'The edge of the $runtimeType that is overflowing has been '
+        'marked in the rendering with a yellow and black striped pattern. This '
+        'is usually caused by the contents being too big for the '
+        '$runtimeType.\n'
+        'This is considered an error condition because it indicates that there '
+        'is content that cannot be seen. If the content is legitimately bigger '
+        'than the available space, consider clipping it with a RectClip widget '
+        'before putting it in the $runtimeType, or using a scrollable '
+        'container, like a ListView, for example.';
+  }
+
+  /// Returns a [RelativeRect] that describes the overflow conditions
+  /// calculated.
+  ///
+  /// Values in the result will be positive if there is an overflow on
+  /// that side.
+  RelativeRect get overflow {
+    _overflow ??= new RelativeRect.fromRect(
+      _overflowContainerRect,
+      _overflowChildRect,
+    );
+    return _overflow;
+  }
+
+  RelativeRect _overflow;
+
+  /// The [Rect] of the child of the container to be checked for
+  /// overflow.
+  ///
+  /// Both [overflowChildRect] and [overflowContainerRect] must be set before
+  /// the overflow will be calculated correctly.
+  Rect get overflowChildRect => _overflowChildRect;
+  set overflowChildRect(Rect overflowChildRect) {
+    if (overflowChildRect == _overflowChildRect)
+      return;
+    _overflowChildRect = overflowChildRect;
+    _markOverflowNeedsUpdate();
+  }
+
+  Rect _overflowChildRect = Rect.zero;
+
+  /// The [Rect] of the container that is to be checked for overflow.
+  ///
+  /// Both [overflowChildRect] and [overflowContainerRect] must be set before
+  /// the overflow will be calculated correctly.
+  Rect get overflowContainerRect => _overflowContainerRect;
```

Comment:
It's weird that overflowChildRect and overflowContainerRect become public mutable properties on the client RenderObject. They should at least be @protected, but maybe it's worth going even further and making them just be arguments to overflowPaintIndicator.